File: 226-Invert-Binary-Tree/226-Invert-Binary-Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        if root == None:
            return None
        def invert(root):
            if root == None:
                return
            temp = root.right
            root.right = root.left
            root.left = temp
            invert(root.left)
            invert(root.right)
        invert(root)
        return root

File: 226-Invert-Binary-Tree/226-Invert-Binary-Tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public TreeNode invertTree(TreeNode root) {
        if (root == null) return null;
        return invert(root);

    }
    private static TreeNode invert(TreeNode root){
        if(root == null) return null;
        if(root.right == null && root.left == null) return root;
        TreeNode right = invert(root.right);
        TreeNode left = invert(root.left);
        TreeNode temp = right;
        root.right = left;
        root.left = temp;
        return root;

    }
}